### ISSUE TITLE
Can't restore DataFrame with pandas.NaT in dtype = object column from parquet

### ISSUE THREAD

**Issue by sniekamp** [JDASoftwareGroup/kartothek#341]
### Problem description

When I store a DataFrame that contains `pandas.NaT` in a column of dtype = object as a parquet file, I can't restore it again.

In my example, there's no good reason to choose dtype = object, but if kartothek doesn't support the roundtrip in this case, it would be preferable if if failed already when trying to store the DataFrame.

### Example code (ideally copy-pastable)

```python
from tempfile import TemporaryDirectory
from storefact import get_store_from_url
import pandas as pd
from kartothek.serialization import ParquetSerializer

dataset_dir = TemporaryDirectory()
store = get_store_from_url(f"hfs://{dataset_dir.name}")

df = pd.DataFrame({
        "time": [pd.NaT]
    },
    dtype=object,
)
ParquetSerializer().store(store, "df", df)
ParquetSerializer().restore_dataframe(store, "df.parquet")
```
```
Traceback (most recent call last):
  File "kartothek_test.py", line 15, in <module>
    ParquetSerializer().restore_dataframe(store, "df.parquet")
  File "/Users/sniekamp/venv/test/lib/python3.6/site-packages/kartothek/serialization/_parquet.py", line 201, in restore_dataframe
    df = table.to_pandas(categories=categories, date_as_object=date_as_object)
  File "pyarrow/array.pxi", line 587, in pyarrow.lib._PandasConvertible.to_pandas
  File "pyarrow/table.pxi", line 1640, in pyarrow.lib.Table._to_pandas
  File "/Users/sniekamp/venv/test/lib/python3.6/site-packages/pyarrow/pandas_compat.py", line 766, in table_to_blockmanager
    blocks = _table_to_blocks(options, table, categories, ext_columns_dtypes)
  File "/Users/sniekamp/venv/test/lib/python3.6/site-packages/pyarrow/pandas_compat.py", line 1102, in _table_to_blocks
    list(extension_columns.keys()))
  File "pyarrow/table.pxi", line 1107, in pyarrow.lib.table_to_blocks
  File "pyarrow/error.pxi", line 85, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Casting from timestamp[us] to timestamp[ns] would result in out of bounds timestamp: -62135596800000000
```
### Used versions

```
attrs==20.1.0
cffi==1.14.2
click==7.1.2
dask==2.24.0
decorator==4.4.2
fsspec==0.8.0
kartothek==3.13.1
locket==0.2.0
milksnake==0.1.5
msgpack==1.0.0
numpy==1.19.1
pandas==1.0.5
partd==1.1.0
prompt-toolkit==3.0.6
pyarrow==0.17.1
pycparser==2.20
python-dateutil==2.8.1
pytz==2020.1
PyYAML==5.3.1
simplejson==3.17.2
simplekv==0.14.1
six==1.15.0
storefact==0.10.0
toolz==0.10.0
uritools==3.0.0
urlquote==1.1.4
wcwidth==0.2.5
zstandard==0.14.0
```

Same behaviour with pyarrow==0.17.0.

With pyarrow==0.16.0, `restore_dataframe` doesn't fail, but the NaT value is converted to a finite timestamp.

**Comment by fjetter** [comment]
Thanks for the report. Good news is that with the latest version of pyarrow (1.0.1) this, at least, doesn't raise any longer but the returned types are not stable.

```python
   ...: df_restored = ParquetSerializer().restore_dataframe(store, "df.parquet")

In [5]: df_restored
Out[5]:
   time
0  None

In [6]: pa.__version__
Out[6]: '1.0.1'

In [7]: df_restored.dtypes
Out[7]:
time    object
dtype: object
```

@xhochy what would you consider the expected behaviour? I'd expect arrow to use the pandas metadata to infer the type if the array is null but that's apparently not the case. It also doesn't look like a classical regression since apparently this behaviour changed multiple times already.

**Comment by fjetter** [comment]
I just noticed that `object` as a return type is indeed type stable since it was cast to object. therefor doesn't seem to be an issue with the most recent pyarrow

**Comment by fjetter** [comment]
Closing since fixed by `pyarrow>=1.0.1`
